service/controller/resource/namespaceauth/create.go

```diff
@@ -0,0 +1,66 @@
+package namespaceauth
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/rbac-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	namespace, err := key.ToNamespace(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	resources, err := r.k8sClient.Discovery().ServerPreferredResources()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	viewAllRole, err := newViewAllRole(resources)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "creating view role %#q in namespace %#q", viewAllRole.Name, namespace.Name)
+
+	_, err = r.k8sClient.RbacV1().Roles(namespace.Name).Get(viewAllRole.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err := r.k8sClient.RbacV1().Roles(namespace.Name).Create(viewAllRole)
+		if err != nil {
```

Comment:
Perhaps just a minor thing, but maybe add handling for `AlreadyExists` here? I'm not sure if that could ever happen, but in case it does, we would still avoid one false alarm from error and be able to continue rest of the code.

Also same thing below for `RoleBinding`.